pkg/template/controller/templateinstance_controller.go

```diff
@@ -97,6 +103,13 @@ func NewTemplateInstanceController(config *rest.Config, kc kclientsetinternal.In
 		},
 	})
 
+	discoveryClient := cacheddiscovery.NewMemCacheClient(c.kc.Discovery())
+	c.dynamicRestMapper = discovery.NewDeferredDiscoveryRESTMapper(discoveryClient, meta.InterfacesForUnstructured)
+	c.dynamicRestMapper.Reset()
```

Comment:
This is the third or fourth one of these I've seen and it also happens to *different*.  Put it on the `ControllerContext` and we can fix the debt on the other ones next time we touch them.  Might even take this upstream now that we all use the same discovery.

Reply:
note to self: david appears to be talking about initializing this in controller_manager.go:newControllerContext().